src/CodeCoverage/Collector.php

```diff
@@ -38,6 +45,72 @@ public static function start($file)
 	}
 
 
+	protected static function getCoverAnnotations()
+	{
+		global $argv;
+		$testFile = $argv[0];
+		return Helpers::parseDocComment(file_get_contents($testFile));
+	}
+
+
+	/**
+	 * @return int|array[] (filename => \Reflector[])
+	 * @throws \ReflectionException
+	 */
+	protected static function getCoverFilters()
+	{
+		$annotations = static::getCoverAnnotations();
+		if (isset($annotations['coversNothing'])) {
+			if (isset($annotations['covers'])) {
+				throw new \Exception('Using both @covers and @coversNothing is not supported');
+			}
+
+			return self::COVER_NOTHING;
+		}
+
+		if (!isset($annotations['covers'])) {
+			// TODO warn user to use covers
+			return self::COVER_ALL;
+		}
+
+		$filters = array();
+		foreach ((array) $annotations['covers'] as $name) {
+			$ref = NULL;
+			try {
+				if (strpos($name, '::') !== FALSE) {
+					$ref = new \ReflectionMethod(rtrim($name, '()'));
+
+				} else {
+					$ref = new \ReflectionClass($name);
+				}
+
+			} catch (\ReflectionException $e) {
+				throw new \Exception("Failed to find '$name' when generating coverage", NULL, $e);
+			}
+
+			$filters[$ref->getFileName()][] = $ref;
+		}
+
+		return $filters;
+	}
+
+
+	/**
+	 * @param \ReflectionClass[]|\ReflectionMethod[] $refs
+	 * @param int                                    $line
+	 * @return bool
+	 */
+	private static function isCovered(array $refs, $line)
```

Comment:
I think it would be better to pass `$filename, $filters, $line` and move a bit more logic from `save` here.

~~~php
if ($filters === self::COVER_ALL) {
     return TRUE;
} elseif ($filters === self::COVER_NOTHING || empty($filters[$filename])) {
    return FALSE;
} else {
    foreach ($filters[$filename] as $ref) {
        // ...
    }
}
~~~